src/irr/asset/CGLIWriter.cpp

```diff
@@ -0,0 +1,359 @@
+			case EF_R8G8B8_SINT: return getTranslatedFinalFormat(FORMAT_RGB8_SINT_PACK8);			//GL_RGB8I
+			case EF_R8G8B8A8_SINT: return getTranslatedFinalFormat(FORMAT_RGBA8_SINT_PACK8);			//GL_RGBA8I
+
+			case EF_R16_SINT: return getTranslatedFinalFormat(FORMAT_R16_SINT_PACK16);				//GL_R16I
+			case EF_R16G16_SINT: return getTranslatedFinalFormat(FORMAT_RG16_SINT_PACK16);			//GL_RG16I
+			///case EF_R16G16B16_SINT: return getTranslatedFinalFormat(FORMAT_RGB16_SINT_PACK16);			//GL_RGB16I
+			case EF_R16G16B16A16_SINT: return getTranslatedFinalFormat(FORMAT_RGBA16_SINT_PACK16);			//GL_RGBA16I
+
+			case EF_R32_SINT: return getTranslatedFinalFormat(FORMAT_R32_SINT_PACK32);				//GL_R32I
+			case EF_R32G32_SINT: return getTranslatedFinalFormat(FORMAT_RG32_SINT_PACK32);			//GL_RG32I
+			///case EF_R32G32B32_SINT: return getTranslatedFinalFormat(FORMAT_RGB32_SINT_PACK32);			//GL_RGB32I
+			case EF_R32G32B32A32_SINT: return getTranslatedFinalFormat(FORMAT_RGBA32_SINT_PACK32);			//GL_RGBA32I
+
+			// Floating formats
+			case EF_R16_SFLOAT: return getTranslatedFinalFormat(FORMAT_R16_SFLOAT_PACK16);				//GL_R16F
+			case EF_R16G16_SFLOAT: return getTranslatedFinalFormat(FORMAT_RG16_SFLOAT_PACK16);			//GL_RG16F
+			///case EF_R16G16B16_SFLOAT: return getTranslatedFinalFormat(FORMAT_RGB16_SFLOAT_PACK16);			//GL_RGB16F
+			case EF_R16G16B16A16_SFLOAT: return getTranslatedFinalFormat(FORMAT_RGBA16_SFLOAT_PACK16);			//GL_RGBA16F
+
+			case EF_R32_SFLOAT: return getTranslatedFinalFormat(FORMAT_R32_SFLOAT_PACK32);				//GL_R32F
+			case EF_R32G32_SFLOAT: return getTranslatedFinalFormat(FORMAT_RG32_SFLOAT_PACK32);			//GL_RG32F
+			///case EF_R32G32B32_SFLOAT: return getTranslatedFinalFormat(FORMAT_RGB32_SFLOAT_PACK32);			//GL_RGB32F
+			case EF_R32G32B32A32_SFLOAT: return getTranslatedFinalFormat(FORMAT_RGBA32_SFLOAT_PACK32);			//GL_RGBA32F
+
+			case EF_R64_SFLOAT: return getTranslatedFinalFormat(FORMAT_R64_SFLOAT_PACK64);			//GL_R64F
+			case EF_R64G64_SFLOAT: return getTranslatedFinalFormat(FORMAT_RG64_SFLOAT_PACK64);		//GL_RG64F
+			///case EF_R64G64B64_SFLOAT: return getTranslatedFinalFormat(FORMAT_RGB64_SFLOAT_PACK64);		//GL_RGB64F
+			case EF_R64G64B64A64_SFLOAT: return getTranslatedFinalFormat(FORMAT_RGBA64_SFLOAT_PACK64);		//GL_RGBA64F
+
+			// sRGB formats
+			case EF_R8_SRGB: return getTranslatedFinalFormat(FORMAT_R8_SRGB_PACK8);				//GL_SR8_EXT
+			case EF_R8G8_SRGB: return getTranslatedFinalFormat(FORMAT_RG8_SRGB_PACK8);				//GL_SRG8_EXT
+			case EF_R8G8B8_SRGB: return getTranslatedFinalFormat(FORMAT_RGB8_SRGB_PACK8);			//GL_SRGB8
+			case EF_R8G8B8A8_SRGB: return getTranslatedFinalFormat(FORMAT_RGBA8_SRGB_PACK8);		//GL_SRGB8_ALPHA8
+
+			// Compressed formats
+			case EF_PVRTC1_4BPP_UNORM_BLOCK_IMG: return getTranslatedFinalFormat(FORMAT_RGBA_PVRTC1_8X8_UNORM_BLOCK32);		//GL_COMPRESSED_RGBA_PVRTC_4BPPV1_IMG
+			case EF_PVRTC1_2BPP_UNORM_BLOCK_IMG: return getTranslatedFinalFormat(FORMAT_RGBA_PVRTC1_8X8_UNORM_BLOCK32);		//GL_COMPRESSED_RGBA_PVRTC_2BPPV1_IMG
```

Comment:
it's wrong for sure, but no idea what it is 
I couldn't find anything about this format

Reply:
which ones are wrong then?